syntax = "proto3";

package syft.lib.torch;

import "proto/lib/python/collections/ordered_dict.proto";
import "proto/core/plan/plan.proto";
import "proto/lib/python/dict.proto";
import "proto/lib/python/list.proto";

message Module {
    string module_type = 1;
    string module_name = 2;
    optional string module_repr = 3;
    string python_module = 4;
    repeated Module children = 5;
    optional syft.lib.python.List module_args = 6;
    optional syft.lib.python.Dict module_kwargs = 7;
    syft.lib.python.collections.OrderedDict state_dict = 8;
    syft.lib.python.collections.OrderedDict parameters = 9;
    optional syft.core.plan.Plan forward = 10;
    optional syft.lib.python.collections.OrderedDict _uid2attr = 11;
}